### PR TITLE
Added FileStructureCommand & API

### PR DESCRIPTION
**Pull request by tsmaeder**
Adds a command to compute a file structure. Also introduced an API project for clients to consume command id's and DTO classes

Fix for https://github.com/eclipse/che/issues/6752

### CODE REVIEW

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -4,12 +4,18 @@ Bundle-Name: jdt.ls.extension Core Extension
...
+ org.eclipse.jdt.core,
  org.eclipse.jdt.launching,
+ org.eclipse.lsp4j,
+ org.eclipse.lsp4j.jsonrpc.json.adapters,
  org.osgi.framework;version="1.3.0"
 Require-Bundle: org.eclipse.core.runtime,
  org.eclipse.jdt.core,
- org.eclipse.jdt.ls.core
+ org.eclipse.jdt.ls.core,
+ org.eclipse.jdt.core.manipulation
```
Comment (benoitf) [inline org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/META-INF/MANIFEST.MF]: would it help to use maven-bundle-plugin and let BND generate the correct import packages.

File: org-eclipse-che-jdt-ls-extension/pom.xml [hunk 2]
```diff
@@ -11,205 +11,211 @@
...
-                <groupId>org.eclipse.tycho</groupId>
-                <artifactId>tycho-compiler-plugin</artifactId>
-                <version>${tycho.version}</version>
-                <configuration>
-                    <useProjectSettings>false</useProjectSettings>
-                    <compilerArgs>
-                        <arg>-err:-forbidden</arg>
-                        <arg>-warn:-forbidden</arg>
-                    </compilerArgs>
+<!-- Copyright (c) 2012-2017 Red Hat, Inc. All rights reserved. This program
```
Comment (benoitf) [inline org-eclipse-che-jdt-ls-extension/pom.xml]: it seems format of the file has been changed (space <--> tabs) so it's hard to check what is different

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/jdt/ls/core/internal/hover/JavaElementLabelComposer.java [hunk 3]
```diff
@@ -0,0 +1,1335 @@
...
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.osgi.util.NLS;
+
+/**
+ * Implementation of {@link JavaElementLabels}.
+ *
+ * @since 3.5
+ */
+public class JavaElementLabelComposer {
+
```
Comment (svor) [inline org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/jdt/ls/core/internal/hover/JavaElementLabelComposer.java]: Why you don't use this class https://github.com/svor/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/hover/JavaElementLabelComposer.java ? As i understand you can add `org.eclipse.jdt.ls.core.internal.hover` to the manifest into Import-Package block and use it.

Comment (tsmaeder) [reply org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/jdt/ls/core/internal/hover/JavaElementLabelComposer.java]: That package is not exported and in consequence, you get a NoClassDefFoundException at runtime.

**Pull request merged**